### DISCUSSION TITLE
No option "Only select repositories" when creating a fine grained acces token.

### DISCUSSION THREAD

**Discussion by ghost** [community/community#73027] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

I am trying to create an access token for a private repo. With the access token, I only want to grant access to this repo, and no others. Online I see that the fine grained access token has an option "Only select repositories", however when I am creating a token this option does not show up (see the screenshot).
How can I get this option?

![image](https://github.com/community/community/assets/125864665/c3e04508-5083-46a8-935d-0edbbf3da610)

edit: In several posts I see users with fine grained access tokens that do get the option for "only select repositories". For example in the GIF on this link:https://github.blog/2022-10-18-introducing-fine-grained-personal-access-tokens-for-github/.

**Comment by b8zeek** [comment]
The option to grant access to select repositories when creating a personal access token is not available. Personal access tokens in GitHub provide access to all repositories the user has access to, based on the selected scopes.

But you can use GitHub Apps, it allows more fine-grained permissions compared to personal access tokens, you can specify which repositories the app can access.

After creating the app, you can install it on your account and give it access to specific repositories. The app will then have its own unique access token which can be used to authenticate on behalf of the app.

Be careful when handling access tokens as they provide access to your repositories; never share them publicly or commit them into your code.

🍀

**Reply by ghost** [reply]
Thank you very much for your reply.

I tried creating the github app, but I still do not get the option to select only a single repository. This is what I get:
![image](https://github.com/community/community/assets/125864665/92188146-e007-4e07-a227-4453005b53ab)

Also, online I see users with fine grained access tokens that do get the option for "only select repositories". For example in the GIF on this link:https://github.blog/2022-10-18-introducing-fine-grained-personal-access-tokens-for-github/.

**Comment by b8zeek** [comment]
All right, here we go...

OAuth GitHub App
If we're speaking about these apps, create an OAuth GitHub app. Next, you need to set the right permissions... First, create an organization and transfer all the repos you want to have accessible. The organization can remain private. After that you need to grant a permission to those repos following the next steps:

GitHub account => Settings => Applications => Authorized OAuth Apps => Click on the application you want to manage => Organization access => Click on "Grant" or "Request" next to the organization you want to give access to. If you're an owner, you'll see "Grant", otherwise you'll see "Request".

GitHub App
The ability to limit access to specific repositories is a feature of GitHub Apps, so if you need to limit access to specific repositories, you might want to consider creating a GitHub App instead of an OAuth App. GitHub Apps offer more granular permissions and can be installed directly on individual repositories.

Hope this is sufficient for you, if so, I would appreciate closing the discussion. 😄 Cheers. 🍺

P.S. Try reading the documentation, you might find the valuable insights. ☺️

🍀

**Comment by jsantos17** [comment, marked as answer]
I find the selected answer unsatisfying. 

This appears to happen when the user has no repositories available. By default, can only generate fine grained access tokens for repositories your user *owns*, so if your user owns no repositories you won't see the option to select repositories. You're likely trying to generate a token for a repository owned by an *organization*, so you need make sure the organization the repo belongs to has enabled "Allow access via fine-grained personal access tokens" in the organizations settings. [Docs here](https://docs.github.com/en/organizations/managing-programmatic-access-to-your-organization/setting-a-personal-access-token-policy-for-your-organization#restricting-access-by-fine-grained-personal-access-tokens).

After this you'll be able to see the option.

**Reply by jhackel-hypo** [reply]
Yes, this is it! Thank you. Probably @Fedde99 might want do select this as the correct answer. Plus, he should point out, that this affects GitHub organization users.

**Reply by ghost** [reply]
Indeed, the issue was that the repository was owned by the organization!

**Reply by dariusonsched** [reply]
The solution posed by @jsantos17 and affirmed by @jhackel-hypo and @Fedde99 does not work in our situation. The organization has enabled "Allowed access via fine-grained personal access tokens" and the user attempting to generate a select-repository fine-grained access token also happens to be an Owner of the organization and has owner+admin rights on every single repository. However, they are not presented with the option to only grant the fine-grained access token on Select Repositories.

**Reply by vedmaka** [reply]
We experience the same as @dariusonsched  - the organisation has the setting enabled and the user creating a fine-grained token is the org owner and repos owner. Only personal repos show up in the dropdown, no orgs repos present.

**Comment by dariusonsched** [comment]
After finding the selected answer incomplete, **I have discovered how to make the option appear _even after you have enabled_** "Allow access via fine-grained personal access tokens" in the [organization settings](https://github.com/organizations/Onsched/settings/personal-access-tokens).

When creating a fine-grained personal access token, the fourth setting field is a drop-down entitled "Resource owner". By default, this will show your user profile name. It is non-intuitive to think you need to change anything here because you are, after all, generating a Personal Access Token.

However, if you **click on the "Resource owner" drop down list**, you will then be able to **select your Organization**. Then, you will be presented with the option "Only select repositories" under Repository access.

Hopefully, this answer helps others.

**Reply by timethernom** [reply]
Oh, I just saw this follow up after replying to your previous comment. Yes, this fixed it for me also.  Thanks!

**Reply by gherbstman** [reply]
Thanks, Selecting Resource Owner was really not obvious.

**Reply by vedmaka** [reply]
@dariusonsched  Thanks!!

**Comment by vincerubinetti** [comment]
Maybe this is obvious to others, but the account trying to create the token must also be a **member** of the org that owns the repo.

In my case, I had a freshly created "bot" account (not associated with anything) that I wanted to use for automated tasks, and I wanted to generate a personal access token that gave it permission to create issues on a particular repo in a particular org. Only after I did all of these things (to sum up all the notes in this discussion) did I get the option to select a specific repository:

1. Make sure `Allow access via fine-grained personal access tokens` is enabled in the org-wide settings
2. Make sure the account trying to create the token is a member of the org that owns the repo
3. When creating the token, select the org, not the username, in the `Resource owner` dropdown
